File: tests/refactor-history/py-sibling-gates-aligned/01-before/gates.py
def require_save_activity(activity):
    assert activity.controller_lambda_observations > 0
    assert 0 < activity.controller_lambda_mean <= 1
    assert activity.input_effect_mean > 0


def verify_release_activity(activity):
    assert activity.controller_lambda_observations > 0
    assert 0 < activity.controller_lambda_mean <= 1
    assert activity.input_effect_mean > 0

File: tests/refactor-history/jl-sibling-gate-asymmetry/02-after/gates.jl
function require_save_activity(activity)
    observations = activity.controller_lambda_observations
    lambda_mean = activity.controller_lambda_mean
    input_mean = activity.input_effect_mean
    observations > 0 || throw(ArgumentError("save activity has no observations"))
    isfinite(lambda_mean) && lambda_mean > 0 ||
        throw(ArgumentError("save lambda mean is invalid"))
    isfinite(input_mean) && input_mean > 0 ||
        throw(ArgumentError("save input mean is invalid"))
    return true
end

function require_resume_activity(activity)
    observations = activity.controller_lambda_observations
    lambda_mean = activity.controller_lambda_mean
    input_mean = activity.input_effect_mean
    isnan(lambda_mean) && iszero(observations) && return true
    observations > 0 || throw(ArgumentError("resume activity has no observations"))
    isfinite(lambda_mean) && lambda_mean > 0 ||
        throw(ArgumentError("resume lambda mean is invalid"))
    isfinite(input_mean) && input_mean > 0 ||
        throw(ArgumentError("resume input mean is invalid"))
    return true
end
